### PR TITLE
Add info about customer number swap in admin

### PR DESCRIPTION
**Pull request by JoshuaBehrens**
### 1. Why is this change necessary?

When your attention leaves the customer create form in the admin right after saving the new entry, you might miss THAT THE CUSTOMER NUMBER JUST CHANGED.
The change itself is very good and important, but that missing alert is quite important.

### 2. What does this change do, exactly?

image

### 3. Describe each step to reproduce the issue or behaviour.

1. Have a moment of high-frequent registrations, that also take place in the admin
2. Be a support agent
3. Create customer accounts via admin
4. Shop just launched
5. Lots of new customers arrive
6. Phone rings 
7. "Hello, this is Telefonman, I go immer ran"
8. In that phone call you pass the customer number to the person calling
9. Save form and say goodbye
10. 3 minutes later
11. Get support case in where a customer claims having a different customer number than told
12. GDPR thoughts arise! \
alarm mode activated
13. WEE WOO! IS THIS CUSTOMER DATA FROM A DIFFERENT CUSTOMER?
14. No it is just Shopware doing Shopware things 🙃

### 4. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/workflow/2020-08-03-implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

<a href="https://gitpod.io/#https://github.com/shopware/platform/pull/2776"></a>

### CODE REVIEW

File: src/Administration/Resources/app/administration/src/module/sw-customer/snippet/de-DE.json [hunk 1]
```diff
@@ -223,6 +223,7 @@
       "buttonEdit": "Bearbeiten",
       "tabGeneral": "Allgemein",
       "tabAddresses": "Adressen",
+      "messageSaveInfoNumberChange": "Der Kunde hat die neue Kundennummer '{newCustomerNumber}' reserviert bekommen. Vorher: '{oldCustomerNumber}'.",
```
Comment (keulinho) [inline src/Administration/Resources/app/administration/src/module/sw-customer/snippet/de-DE.json]:
```suggestion
      "messageSaveInfoNumberChange": "Der Kunde hat die neu generierte Kundennummer '{newCustomerNumber}' bekommen. Vorher: '{oldCustomerNumber}'.",
```
is more clearer IMHO, and also in line with the english translation. Also maybe a hint on why this is happening may make sense, something like:
```
Due to multiple simultaneous registration, the customer got a newly generated customer number.
```

What do you think?

File: src/Administration/Resources/app/administration/src/module/sw-customer/page/sw-customer-create/index.js [hunk 2]
```diff
@@ -160,6 +162,17 @@ Component.register('sw-customer-create', {
                     this.customerRepository.save(this.customer).then(() => {
                         this.isLoading = false;
                         this.isSaveSuccessful = true;
+
+                        const newCustomerNumber = this.customer.customerNumber;
+
+                        if (oldCustomerNumber !== newCustomerNumber) {
+                            this.createNotificationInfo({
```
Comment (keulinho) [inline src/Administration/Resources/app/administration/src/module/sw-customer/page/sw-customer-create/index.js:169]: can you add a jest test that verifies that the notification is shown if the customer number changes?

Comment (JoshuaBehrens) [reply src/Administration/Resources/app/administration/src/module/sw-customer/page/sw-customer-create/index.js:169]: One test this by mocking the mixin, right?

File: src/Administration/Resources/app/administration/src/module/sw-customer/snippet/de-DE.json (hunk 1 above)
Comment (JoshuaBehrens) [reply src/Administration/Resources/app/administration/src/module/sw-customer/snippet/de-DE.json]: I know my text was more technical. On the other side your text is also a bit too narrowed down to registration. Would you count a customer import via API a "registration" for an admin user?

File: src/Administration/Resources/app/administration/src/module/sw-customer/page/sw-customer-create/index.js (hunk 2 above)
Comment (keulinho) [reply src/Administration/Resources/app/administration/src/module/sw-customer/page/sw-customer-create/index.js:169]: exactly, just making sure createNotififcation is only called when the customer number changed

File: src/Administration/Resources/app/administration/src/module/sw-customer/snippet/de-DE.json (hunk 1 above)
Comment (JoshuaBehrens) [reply src/Administration/Resources/app/administration/src/module/sw-customer/snippet/de-DE.json]: Please review new texts

Comment (keulinho) [reply src/Administration/Resources/app/administration/src/module/sw-customer/snippet/de-DE.json]: LGTM :+1:

**Comment by keulinho** [comment]
@JoshuaBehrens the jest test is still missing, i'm closing this for inactivity now, feel free to reopen when you get to work on it again :v:

**Pull request closed without merging**